src/Components/Server/src/Builder/ComponentEndpointRouteBuilderExtensions.cs

```diff
@@ -113,6 +115,33 @@ public static ComponentEndpointConventionBuilder MapBlazorHub(this IEndpointRout
             endpoints.CreateApplicationBuilder().UseMiddleware<CircuitJavaScriptInitializationMiddleware>().Build())
             .WithDisplayName("Blazor initializers");
 
-        return new ComponentEndpointConventionBuilder(hubEndpoint, disconnectEndpoint, jsInitializersEndpoint);
+        var blazorEndpoint = GetBlazorEndpoint(endpoints);
+
+        return new ComponentEndpointConventionBuilder(hubEndpoint, disconnectEndpoint, jsInitializersEndpoint, blazorEndpoint);
+    }
+
+    private static IEndpointConventionBuilder GetBlazorEndpoint(IEndpointRouteBuilder endpoints)
+    {
+        var options = new StaticFileOptions
+        {
+            FileProvider = new ManifestEmbeddedFileProvider(typeof(ComponentEndpointRouteBuilderExtensions).Assembly)
+        };
+
+        var app = endpoints.CreateApplicationBuilder();
+        app.Use(next => context =>
+        {
+            // Set endpoint to null so the static files middleware will handle the request.
+            context.SetEndpoint(null);
+
+            return next(context);
+        });
+        app.UseStaticFiles(options);
+
+        var blazorEndpoint = endpoints.Map("/_framework/blazor.server.js", app.Build())
+            .WithDisplayName("Blazor static files");
```

Comment:
This approach looks fairly plausible but would love to get @javiercn's take since he has focused a lot on how the server-side endpoints are set up and how static files are served.

Reply:
This will work, the only "slight" concern that I have is that we loose the ability to server blazor.server.js independently of where we are serving the other endpoints, but I do not see a big drawback to it.

Essentially if you want to map blazor to a subdir, you need to call `app.MapPath("subdir", sub => { sub.UseRouting(); sub.UseStaticFiles(); sub.UseEndpoints(e => e.MapComponentHub()); }`, set the base path on the document accordingly.

There might have been other ways to make it work that might not work now, but I do not think they are critical.